packages/@aws-cdk/core/lib/cfn-codedeploy-blue-green-hook.ts

```diff
@@ -0,0 +1,511 @@
+
+  /**
+   * The traffic routing configuration.
+   */
+  readonly trafficRouting: CfnTrafficRouting;
+}
+
+/**
+ * The application actually being deployed.
+ * Type of the {@link CfnCodeDeployBlueGreenHookProps.applications} property.
+ */
+export interface CfnCodeDeployBlueGreenApplication {
+  /**
+   * The target that is being deployed.
+   */
+  readonly target: CfnCodeDeployBlueGreenApplicationTarget;
+
+  /**
+   * The detailed attributes of the deployed target.
+   */
+  readonly ecsAttributes: CfnCodeDeployBlueGreenEcsAttributes;
+}
+
+/**
+ * Construction properties of {@link CfnCodeDeployBlueGreenHook}.
+ */
+export interface CfnCodeDeployBlueGreenHookProps {
+  /**
+   * The IAM Role for CloudFormation to use to perform blue-green deployments.
+   */
+  readonly serviceRole: string;
+
+  /**
+   * Properties of the Amazon ECS applications being deployed.
+   */
+  readonly applications: CfnCodeDeployBlueGreenApplication[];
+
+  /**
+   * Traffic routing configuration settings.
+   *
+   * @default - time-based canary traffic shifting, with a 15% step percentage and a five minute bake time
+   */
+  readonly trafficRoutingConfig?: CfnTrafficRoutingConfig;
+
+  /**
+   * Additional options for the blue/green deployment.
+   *
+   * @default - no additional options
+   */
+  readonly additionalOptions?: CfnCodeDeployBlueGreenAdditionalOptions;
+
+  /**
+   * Use lifecycle event hooks to specify a Lambda function that CodeDeploy can call to validate a deployment.
+   * You can use the same function or a different one for deployment lifecycle events.
+   * Following completion of the validation tests,
+   * the Lambda {@link CfnCodeDeployBlueGreenLifecycleEventHooks.afterAllowTraffic}
+   * function calls back CodeDeploy and delivers a result of 'Succeeded' or 'Failed'.
+   *
+   * @default - no lifecycle event hooks
+   */
+  readonly lifecycleEventHooks?: CfnCodeDeployBlueGreenLifecycleEventHooks;
+}
+
+/**
+ * A CloudFormation Hook for CodeDeploy blue-green ECS deployments.
+ *
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/blue-green.html#blue-green-template-reference
+ */
+export class CfnCodeDeployBlueGreenHook extends CfnHook {
+  /**
+   * A factory method that creates a new instance of this class from an object
+   * containing the CloudFormation properties of this resource.
+   * Used in the @aws-cdk/cloudformation-include module.
+   *
+   * @internal
+   */
+  public static _fromCloudFormation(scope: Construct, id: string, hookAttributes: any,
+    options: FromCloudFormationOptions): CfnCodeDeployBlueGreenHook {
+
+    hookAttributes = hookAttributes || {};
+    const hookProperties = options.parser.parseValue(hookAttributes.Properties);
+    return new CfnCodeDeployBlueGreenHook(scope, id, {
+      serviceRole: hookProperties?.ServiceRole,
+      applications: (hookProperties?.Applications ?? []).map((app: any) => {
+        const target = findResource(app?.Target?.LogicalID);
```

Comment:
`~` This whole ~30 line lambda could probably be its own helper function.

Reply:
Done!